girder_worker/plugins/docker/__init__.py

```diff
@@ -1,12 +1,38 @@
+import subprocess
+
+
 def before_run(e):
     import executor
     if e.info['task']['mode'] == 'docker':
         executor.validate_task_outputs(e.info['task_outputs'])
 
 
+def cleanup(e):
+    """
+    Since files written by docker containers are owned by root, we can't
+    clean them up in the worker process since that typically doesn't run
+    as root. So, we run a lightweight container to clean up the temp dir.
+    """
+    from .executor import DATA_VOLUME
+    if e.info['task']['mode'] == 'docker' and '_tempdir' in e.info['kwargs']:
+        tmpdir = e.info['kwargs']['_tempdir']
+        cmd = [
+            'docker', 'run', '--rm', '-v', '%s:%s' % (tmpdir, DATA_VOLUME),
+            'busybox', 'rm', '-rf', DATA_VOLUME
```

Comment:
Well, if *that* got compromised in the same way, then instead of deleting our root filesystem, we've just given write permission to every user on every file... I'd rather just have them delete the files since it's an ephemeral processing node.